libraries/apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/network/NetworkTransport.kt

```diff
@@ -1,12 +1,20 @@
 package com.apollographql.apollo3.network
 
-import com.apollographql.apollo3.api.Operation
 import com.apollographql.apollo3.api.ApolloRequest
 import com.apollographql.apollo3.api.ApolloResponse
+import com.apollographql.apollo3.api.Operation
+import com.apollographql.apollo3.exception.ApolloException
 import kotlinx.coroutines.flow.Flow
 
 interface NetworkTransport {
 
+  /**
+   * Execute a request.
+   *
+   * @param request the request to execute.
+   * @return a flow of responses.
+   * [ApolloException]s must not be thrown from the flow. Instead, they must be emitted as [ApolloResponse.exception].
```

Comment:
```suggestion
   * This function must not throw in normal circumstances. Instead, return an [ApolloResponse] with a non
   * null [ApolloResponse.exception].
   * It may throw if the error is the result of a programming error like wrongly configured interceptors.
```